pricegraph/src/orderbook.rs

```diff
@@ -478,6 +553,97 @@ mod tests {
         assert!(!orderbook.is_overlapping());
     }
 
+    #[test]
+    fn fills_market_order_with_correct_price() {
+        //    /-101.0--v
+        //   /--105.0--v
+        //  /---111.0--v
+        // 1           2
+        // ^--.0101---/
+        // ^--.0105--/
+        // ^--.0110-/
+        let mut orderbook = orderbook! {
+            users {
+                @1 {
+                    token 1 => 1_000_000,
+                    token 2 => 100_000_000,
+                }
+                @2 {
+                    token 1 => 1_000_000,
+                    token 2 => 100_000_000,
+                }
+                @3 {
+                    token 1 => 1_000_000,
+                    token 2 => 100_000_000,
+                }
+            }
+            orders {
+                owner @1 buying 1 [1_000_000] selling 2 [99_000_000],
```

Comment:
I didn't notice this in the PRs before, but this DSL is really fancy.

Reply:
It makes writing the orderbooks easier and more expressive.